command/operator_config_encryptdecrypt.go

```diff
@@ -0,0 +1,186 @@
+  with {{%s()}} as appropriate. This can only be used with string parameters, and
+  the markers must be inside the quote marks delimiting the string; as an example:
+
+      foo = "{{encrypt(bar)}}"
+
+  By default this will print the new configuration out. To overwrite into the same
+  file use the -overwrite flag.
+
+      $ vault operator config %s -overwrite config.hcl
+`
+	helpText = fmt.Sprintf(helpText, strings.ToLower(subCmd), subCmd, strings.ToLower(subCmd), strings.ToLower(subCmd))
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *OperatorConfigEncryptDecryptCommand) Flags() *FlagSets {
+	set := c.flagSet(0)
+
+	f := set.NewFlagSet("Command Options")
+
+	f.BoolVar(&BoolVar{
+		Name:   "overwrite",
+		Target: &c.flagOverwrite,
+		Usage:  "Overwrite the existing file.",
+	})
+
+	f.BoolVar(&BoolVar{
+		Name:   "strip",
+		Target: &c.flagStrip,
+		Usage:  "Strip the declarations from the file afterwards.",
+	})
+
+	return set
+}
+
+func (c *OperatorConfigEncryptDecryptCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictAnything
+}
+
+func (c *OperatorConfigEncryptDecryptCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *OperatorConfigEncryptDecryptCommand) Run(args []string) (ret int) {
+	op := "decrypt"
+	if c.encrypt {
+		op = "encrypt"
+	}
+
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	path := ""
+	args = f.Args()
+	switch len(args) {
+	case 1:
+		path = strings.TrimSpace(args[0])
+	default:
+		c.UI.Error(fmt.Sprintf("Incorrect arguments (expected 1, got %d)", len(args)))
+		return 1
+	}
+
+	if path == "" {
+		c.UI.Error("A configuration file must be specified")
+		return 1
+	}
+
+	kmses, err := configutil.LoadConfigKMSes(path)
+	if err != nil {
+		c.UI.Error(fmt.Errorf("Error loading configuration from %s: %w", path, err).Error())
+		return 1
+	}
+
+	var kms *configutil.KMS
+	for _, v := range kmses {
+		if strutil.StrListContains(v.Purpose, "config") {
+			if kms != nil {
+				c.UI.Error("Only one seal/kms block marked for \"config\" purpose is allowed")
+				return 1
+			}
+			kms = v
+		}
+	}
+	if kms == nil {
+		c.UI.Error("No seal/kms block with \"config\" purpose defined in the configuration file")
+		return 1
+	}
+
+	d, err := ioutil.ReadFile(path)
+	if err != nil {
+		c.UI.Error(fmt.Errorf("Error reading config file: %w", err).Error())
+		return 1
+	}
+
+	raw := string(d)
+
+	wrapper, err := configutil.ConfigureWrapper(kms, nil, nil, nil)
+	if err != nil {
+		c.UI.Error(fmt.Errorf("Error creating kms: %w", err).Error())
+		return 1
+	}
+
+	wrapper.Init(context.Background())
+	defer wrapper.Finalize(context.Background())
+
+	raw, err = configutil.EncryptDecrypt(raw, !c.encrypt, c.flagStrip, wrapper)
+	if err != nil {
+		c.UI.Error(fmt.Errorf("Error %sing via kms: %w", op, err).Error())
+		return 1
+	}
+
+	if !c.flagOverwrite {
+		c.UI.Output(raw)
+		return 0
+	}
+
+	file, err := os.Create(path)
```

Comment:
Might be handy to have an --output flag (optional) so one could transform into a target file without overwrite.  Or is the intent merely redirecting stdout to said file.